src/khoj/configure.py

```diff
@@ -48,6 +49,7 @@ def configure_server(args, required=False):
     # Initialize the search type and model from Config
     state.search_index_lock.acquire()
     state.SearchType = configure_search_types(state.config)
+    state.model = SearchModels()
```

Comment:
I added this so that the state of models can be reset between configuration steps. This helps in case one of the configuration steps fails below, we won't send stale data.

Reply:
Is this required because of the Github plugin setup being wrapped in a try/except below? A failure to configure the other content types should fail the application from my understanding. Should we let the Github plugin setup fail the application in case it errors out?

Reply:
My initial thought was that it's better to fail gracefully and let the other indexing continue while making the user aware in some other medium that failure occurred. Especially for less-technical folks, might be irksome if half your functionality breaks. I'm fine either way ultimately.

Reply:
But yes, correct. That is why the reset was required. Otherwise, because we fail only silently, it has cached data from the previous successful run.